src/Cameras/XR/babylon.webXRExperienceHelper.ts

```diff
@@ -0,0 +1,116 @@
+
+        private _outputCanvas: HTMLCanvasElement;
+        private _outputCanvasContext: WebGLRenderingContext;
+
+        /**
+         * Creates a WebXRExperienceHelper
+         * @param scene The scene the helper should be created in
+         */
+        constructor(private scene: BABYLON.Scene) {
+            this.camera = new BABYLON.WebXRCamera("", scene);
+            this._sessionManager = new BABYLON.WebXRSessionManager(scene);
+            this.container = new AbstractMesh("", scene);
+            this._sessionManager.initialize();
+        }
+
+        /**
+         * Exits XR mode and returns the scene to its original state
+         * @returns promise that resolves after xr mode has exited
+         */
+        public exitXR() {
+            this.isInStateTransition = true;
+            return this._sessionManager.exitXR();
+        }
+
+        /**
+         * Enters XR mode (This must be done within a user interaction in most browsers eg. button click)
+         * @param sessionCreationOptions options for the XR session
+         * @param frameOfReference frame of reference of the XR session
+         * @returns promise that resolves after xr mode has entered
+         */
+        public enterXR(sessionCreationOptions: XRSessionCreationOptions, frameOfReference: string) {
+            this.isInStateTransition = true;
+
+            this._createCanvas();
+            if (!sessionCreationOptions.outputContext) {
+                sessionCreationOptions.outputContext = this._outputCanvasContext;
+            }
+
+            return this._sessionManager.enterXR(sessionCreationOptions, frameOfReference).then(() => {
+                // Cache pre xr scene settings
+                this._originalSceneAutoClear = this.scene.autoClear;
+                this._nonVRCamera = this.scene.activeCamera;
+
+                // Overwrite current scene settings
+                this.scene.autoClear = false;
+                this.scene.activeCamera = this.camera;
+
+                this._sessionManager.onXRFrameObservable.add(() => {
+                    this.camera.updateFromXRSessionManager(this._sessionManager);
+                });
+
+                this._sessionManager.onXRSessionEnded.addOnce(() => {
+                    // Reset camera rigs output render target to ensure sessions render target is not drawn after it ends
+                    this.camera.rigCameras.forEach((c) => {
+                        c.outputRenderTarget = null;
+                    });
+
+                    // Restore scene settings
+                    this.scene.autoClear = this._originalSceneAutoClear;
+                    this.scene.activeCamera = this._nonVRCamera;
+                    this._sessionManager.onXRFrameObservable.clear();
+                    this._removeCanvas();
+
+                    this.isInStateTransition = false;
+                    this.isInXRMode = false;
+                });
+                this.isInStateTransition = false;

```

Comment:
I think we should add observables to track the xr lifetime

Reply:
good, idea I also removed isInStateTransition  and isInXRMode in favor of a single state enum